### DISCUSSION TITLE
Rely on Exclusion list for filtering. Synchronise everything option.

### DISCUSSION THREAD

**Discussion by HighPriest** [ObsidianPublisher/obsidian-github-publisher#207] [Q&A — Plugin] [answered]
Hi.
Would it fit your goals with this tool, to do not check for the `share key` if the config field is empty and rely on Exclusion list for files we do not want synchronised instead?

It would fit me much better if I could store my public files in one dir, private files in another.
If I work on something I do not want synchronised, I would rather give it a easily distinguishible `[DRAFT]` prefix to mark something not yet ready, than add unnecessary clutter to frontmatter.

So, to summarise. What I believe this requires are two things:
- Add regex support in Plugin Settings/Excluded Folder
- Add omitting of `share key` check if the field is empty

**Comment by Mara-Li** [comment]
I'm sorry, it can't possible, the code rely on this keys for everything. It has a lot of check to prevent mistake, so removing it is not an easy things.
For something like that, you could use a tool to add the share key everywhere & use templater to add it automatically.

For the regex support in excluded folder, I can add it!

**Reply by Mara-Li** [reply]
EDIT: 
![image](https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/30244939/45423d6d-cf89-41ee-8840-9b84a41df201)
Regex should be already supported!

**Reply by Mara-Li** [reply, marked as answer]
See #210
